wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -1297,6 +1294,52 @@ public List<DFULogicalFileWrapper> searchFiles(String logicalFilename, String cl
         return result;
     }
 
+
+    /**
+     * searchFiles
+     *
+     * @param request - the search request
+     * @return - collection of files matching the request passed in
+     * @throws java.lang.Exception general exception
+     * @throws org.hpccsystems.ws.client.wrappers.ArrayOfEspExceptionWrapper array of esp exception wrapper
+     */
+    public List<DFULogicalFileWrapper> searchFiles(DFUQueryRequestWrapper request) throws Exception, ArrayOfEspExceptionWrapper
+    {
+        verifyStub(); // Throws exception if stub failed
+
+        if (request.getLogicalName() != null && request.getLogicalName().startsWith("~"))
+            request.setLogicalName(request.getLogicalName().substring(1));
+
+        DFUQueryResponse resp = null;
+
+        try
+        {
+            resp = ((WsDfuStub) stub).dFUQuery(request.getRaw());
+        }
+        catch (RemoteException e)
+        {
+            throw new Exception("HPCCWsDFUClient.searchFiles encountered RemoteException.", e);
+        }
+        catch (EspSoapFault e)
+        {
+            handleEspSoapFaults(new EspSoapFaultWrapper(e), "Could Not SearchFiles");
+        }
+
+        if (resp.getExceptions() != null) handleEspExceptions(new ArrayOfEspExceptionWrapper(resp.getExceptions()), "Could Not SearchFiles");
+
+        List<DFULogicalFileWrapper> result = new ArrayList<DFULogicalFileWrapper>();
+        ArrayOfDFULogicalFile logicalfilearray = resp.getDFULogicalFiles();
+        if (logicalfilearray != null)
+        {
+            DFULogicalFile[] dfulogicalfilearray = resp.getDFULogicalFiles().getDFULogicalFile();
```

Comment:
we could use logicalfilearray  instead of calling resp.getDFULogicalFiles() again

Reply:
Fixed